PROTOCOL.md

```diff
@@ -1950,6 +1951,18 @@ When `delta.identity.allowExplicitInsert` is false, writers should meet the foll
 - Overflow when calculating generated Identity values should be detected and such writes should not be allowed.
 - `delta.identity.highWaterMark` should be updated to the new highest value when the write operation commits.
 
+## Materialize Partition Columns
+
+Delta supports materializing partition columns in data files. When this feature is enabled, partition columns are physically written to Parquet files alongside the data columns, which can improve query performance for certain workloads, and improve compatibility with external readers, and make these data files easier to interpret for readers unfamiliar with partition values. To support this feature:
+ - The table must be on Writer Version 7, and a feature name `materializePartitionColumns` must exist in the table `protocol`'s `writerFeatures`.
+
+When supported:
+ - The table respect metadata property `delta.enableMaterializePartitionColumnsFeature` for enablement of this feature. The writer feature `materializePartitionColumns` is auto-enabled when this property is set to `true`.
+ - When the writer feature `materializePartitionColumns` is set in the protocol, writers must write partition column values to all newly created parquet data files.
```

Comment:
Let's mimic the wording from IcebergCompatV1 and mention that these partition columns should be after all other columns. For reference, this is the wording under that feature:

`Require that partition column values are materialized into any Parquet data file that is present in the table, placed after the data columns in the parquet schema`